config/config.go

```diff
@@ -545,6 +545,18 @@ func ParseConfigFile(filename string, terragruntOptions *options.TerragruntOptio
 		return nil, err
 	}
 
+	// Initial evaluation of configuration to load flags like IamRole which will be used for final parsing
+	// https://github.com/gruntwork-io/terragrunt/issues/667
+	if include == nil && terragruntOptions.IamRole == "" {
```

Comment:
I think you need to omit the `include == nil` check here, as this means it won't handle a config where the `iam_role` is defined on the child config that has an `include` defined.